sql/lens/top100k/blink_timeseries.sql

```diff
@@ -0,0 +1,33 @@
+SELECT
+  f.yyyymmdd,
+  f.client,
+  f.id,
+  f.feature,
+  f.type,
+  COUNT(1) as num_urls,
+  MAX(total),
+  COUNT(1) / max(total) as num_urls_pct
+FROM
+  `httparchive.blink_features.features` f,
+  `httparchive.summary_pages.*` sp,
```

Comment:
You recently worked on making timeseries queries incremental, eg `summary_pages.${YYYY_MM_DD}_*`. Is there a way to use that pattern here?

Reply:
We could but it doesn’t make a difference as not clustered by date and only one table. So full cost each time whether doing by date or not so figured I’d keep it simple.

Reply:
Wouldn't it still help reduce costs of querying `summary_pages`? Also `blink_features.features` should probably be clustered/partitioned by date.

Reply:
`summary_pages` seems really cheap to query for the columns we need. Particularly compared to `blink_features.features` so didn’t see any saving.

Do agree it should be clustered by date and then worth doing this. Will have a lot at what’s involved but does get complex. Currently the blink features and crux queries are only ones that require different lens SQL so adding date to it is another level of complexity in top of that for no gain at present. But let me see what’s involved.

Reply:
OK I've added this. Does make the script more complicated but blink table is quite big and expensive to query so would be nice to make this cheaper in future if ever clustered/partitioned by date.

Reply:
I've updated the features table schema to use partitioning. It also changes the type of the `yyyymmdd` field from string to date, so querying it would be like `WHERE yyyymmdd = '2021-07-01'` as opposed to `WHERE yyyymmdd = '20210701'`.

Starting with the July crawl, the features table should include rank data per URL. We would need to backfill the rank data for older crawls, but this is a good start.

Reply:
> I've updated the features table schema to use partitioning.

Nice one! Can see it is cheaper.

> I've updated the features table schema to use partitioning. It also changes the type of the yyyymmdd field from string to date, so querying it would be like WHERE yyyymmdd = '2021-07-01' as opposed to WHERE yyyymmdd = '20210701'.

OK changed the queries in https://github.com/HTTPArchive/bigquery/pull/121/commits/0b160ddd84820be82656e3ffa03d596f62306e82.

> Starting with the July crawl, the features table should include rank data per URL. We would need to backfill the rank data for older crawls, but this is a good start.

OK have stuck with also using the `summary_pages` for now until we backfill May and June. Shall we do that before merging this to make the queries cleaner?

Reply:
I'd suggest cleaning up the queries now and running it on the latest dataset. It won't pick up ranking data for May or June until we backfill it, but that's ok.

Reply:
Only issue with that is we have a “trendline” made of one data point. Which ain’t much of a trend line! Was already a little nervous launching with only three data points.

So I can see four options:
1) Backfill. Will have a look at how difficult that is. Can’t imagine the SQL is too hard, though not sure if you can just run an update statement in BigQuery under my HTTPArchive account or need to run in a special way from the command line?
2) Run with old query (like I did for 10k and 1k), then update and merge and use new way for incremental updates going forward. Does mean we can’t rerun these from scratch until we backfill, but hopefully by time we need to do that we’ll have either backfilled or built up a trendline.
3) Wait to turn this on, on website for a few mo this.
4) Live with it just being one data point for now.

Would prefer 1) to be honest so will have a look.

Reply:
Yeah you can write an `UPDATE` statement in the BigQuery UI to run the backfill. Let me know if you run into any issues.

Reply:
I ran this 4 times (for desktop and mobile and May and June) and it seemed to do the trick:

```sql
UPDATE `httparchive.blink_features.features` AS bf
SET bf.rank = sp.rank
FROM `httparchive.summary_pages.2021_06_01_desktop` AS sp
WHERE bf.url = sp.url
AND bf.client = 'desktop'
AND bf.yyyymmdd = '2021-06-01'
and bf.rank is null
```

About 200 are still null but can live with that.

Reply:
Queries have been updated to use new ranking column and top1k run has been completed to test this. think we're good to merge now so will go ahead and do that and then test on the final top1m lens.